lib/github_changelog_generator/project_name_finder.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+module GitHubChangelogGenerator
+  class ProjectNameFinder
```

Comment:
I'm also not excited about placing the test data (the default) inside this implementation.

I'll add "turning the default user/project repo slug into some kind of test setup".